helix-core/src/main/java/org/apache/helix/manager/zk/ZkBaseDataAccessor.java

```diff
@@ -198,21 +271,17 @@ public ZkBaseDataAccessor(String zkAddress, ZkSerializer zkSerializer,
    * @param zkAddress
    * @param pathBasedZkSerializer
    * @param zkClientType
+   *
+   * @deprecated it is recommended to use the builder constructor {@link Builder}
    */
-  public ZkBaseDataAccessor(String zkAddress, PathBasedZkSerializer pathBasedZkSerializer,
+  @Deprecated
+  public ZkBaseDataAccessor(String zkAddress,
+      org.apache.helix.zookeeper.zkclient.serialize.PathBasedZkSerializer pathBasedZkSerializer,
```

Comment:
why do we need the long package name for this serializer?

Reply:
It is to reduce duplicate code for 2 constructors that have serializer parameters. `BasicZkSerializer` implements the long path `PathBasedZkSerializer` interface in zookeeper module. So we need this long path zookeeper `PathBasedZkSerializer`. 
```
public ZkBaseDataAccessor(String zkAddress, ZkSerializer zkSerializer, ZkClientType zkClientType) {
  this(zkAddress, new BasicZkSerializer(zkSerializer), zkClientType);
}
```
Cons: `new BasicZkSerializer(zkSerializer)` just follows the behavior in ZkClientConfig. If it is changed in ZkClientConfig to use another instance other than BasicZkSerializer, the one here would not follow that change. 
Well, if we don't want to reduce the duplicate code, we can just keep the original code.

Reply:
I'd prefer to leave it as is and tackle this later. This sort of refactoring doesn't have to be in the scope of this PR. Having 2 constructors is fine for now.